### PR TITLE
Added new function for merging PodSpec

### PR DESCRIPTION
**Pull request by rahulchheda**
Signed-off-by: Rahul M Chheda <rchheda@infracloud.io>

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [x] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #XXX

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: pkg/kube/pod.go [hunk 1]
```diff
@@ -314,6 +314,23 @@ func CreateAndMergeJsonPatch(original, override crv1alpha1.JSONMap) (crv1alpha1.
...
+
+	// Convert merged json to map[string]interface{}
+	var merged map[string]interface{}
+	err = json.Unmarshal(mergedPatch, &merged)
+	if err != nil {
+		return nil, err
+	}
+	return merged, err
+}
+
```
Comment (PrasadG193) [inline pkg/kube/pod.go:333]: Can we add tests for this?

**Review by PrasadG193 — APPROVED** [review]

**Comment by PrasadG193** [comment]
@rahulchheda let's not commit changes in `go.sum` to avoid conflicts

**Comment by rahulchheda** [comment]
Sure, let me just run go mod tidy

**Comment by PrasadG193** [comment]
I am sure it will make things worse :D. I would suggest reverting changes in go.sum

**Review by pavannd1 — CHANGES_REQUESTED** [review]
Does not work!

**Pull request closed without merging**